app/screens/Video/index.js

```diff
@@ -1,57 +1,90 @@
-  const { languageCode, languageName } = props
+  const { languageCode } = props;
   const [message, setMessage] = useState("");
-  const getUniqueListBy = (arr, key) => {
-    return [...new Map(arr.map(item => [item[key], item])).values()]
-  }
-
-  useEffect(async () => {
+  const fetchVideo = async () => {
+    setIsLoading(true);
     const videosRes = await vApi.get("videos?language=" + languageCode);
+    let videoBook = [];
     let videoAll = [];
-    if (videosRes.success === false) {
-      setMessage("No Video for " + languageName);
-      return;
-    }
-    const videos = videosRes[0].books
-    for (var book in videos) {
-      for (var j = 0; j < videos[book].length; j++) {
-        videoAll.push({
-          title: videos[book][j].title,
-          url: videos[book][j].url,
-          description: videos[book][j].description,
-          theme: videos[book][j].theme,
-          book: book
-        });
+    let found = false;
+    if (videosRes.length > 0) {
+      for (var key in videosRes[0].books) {
+        if (key == bookId) {
+          for (var i = 0; i < videosRes[0].books[key].length; i++) {
+            videoBook.push({
+              title: videosRes[0].books[key][i].title,
+              url: videosRes[0].books[key][i].url,
+              description: videosRes[0].books[key][i].description,
+              theme: videosRes[0].books[key][i].theme,
+            });
+            found = true;
+          }
+          // }
+        } else {
+          for (var j = 0; j < videosRes[0].books[key].length; j++) {
+            videoAll.push({
+              title: videosRes[0].books[key][j].title,
+              url: videosRes[0].books[key][j].url,
+              description: videosRes[0].books[key][j].description,
+              theme: videosRes[0].books[key][j].theme,
+            });
+          }
+        }
       }
-    }
-    var elements = getUniqueListBy(videoAll, "title")
-    if (bookId in videos) {
-      setVideos(videoAll.filter((ele) => ele.book === bookId));
-      setMessage("");
-    } else {
-      if (bookId) {
-        Toast.show({
-          text: `Video for ${bookName} is unavailable. You can check other books`,
-          duration: 8000,
-          position: "top",
-        });
+      if (found) {
+        setVideos(videoBook);
+        setMessage("");
+        setIsLoading(false);
+      } else {
+        if (bookId) {
+          // ToastAndroid.showWithGravityAndOffset(
+          //   'Video for ' + this.state.bookName + ' is unavailable. You can check other books',
+          //   ToastAndroid.LONG,
+          //   ToastAndroid.CENTER,
+          //   25,
+          //   50
+          // );
+          Toast.show({
+            text:
+              "Video for " +
+              bookName +
+              " is unavailable. You can check other books",
+            duration: 8000,
+            position: "top",
+          });
+        }
+        var elements = videoAll.reduce(function (previous, current) {
```

Comment:
Before settings videosAll/elements if we can also sort it by bookid ie gen(1) to rev(66) it would be nice. If there are any issues just leave it for now we can do it in the next release. 1.3.6